packages/protocol/contracts/L1/libs/LibEthDepositing.sol

```diff
@@ -0,0 +1,122 @@
+// SPDX-License-Identifier: MIT
+//  _____     _ _         _         _
+// |_   _|_ _(_) |_____  | |   __ _| |__ ___
+//   | |/ _` | | / / _ \ | |__/ _` | '_ (_-<
+//   |_|\__,_|_|_\_\___/ |____\__,_|_.__/__/
+
+pragma solidity ^0.8.18;
+
+import {LibAddress} from "../../libs/LibAddress.sol";
+import {AddressResolver} from "../../common/AddressResolver.sol";
+import {
+    SafeCastUpgradeable
+} from "@openzeppelin/contracts-upgradeable/utils/math/SafeCastUpgradeable.sol";
+import {TaikoData} from "../TaikoData.sol";
+
+library LibEthDepositing {
+    using LibAddress for address;
+    using SafeCastUpgradeable for uint256;
+
+    // When numEthDepositPerBlock is 32, the average gas cost per
+    // EthDeposit is about 2700 gas. We use 21000 so the proposer may
+    // earn a small profit if there are 32 deposits included
+    // in the block; if there are less EthDeposit to process, the
+    // proposer may suffer a loss so the proposer should simply wait
+    // for more EthDeposit be become available.
```

Comment:
It's hard to guarantee  both the processing time and a low fee. An alternative is to process depsosits only if there are no less than 8 or 16 in the queue, otherwise, skip all. Then it's up to the users to choose between this feature and the bridge.

The break even point shall be much smaller than 32. I'll calculate it.

Reply:
It takes 5 deposits to break even.